packages/walletconnect-solana/src/adapter.ts

```diff
@@ -76,7 +76,7 @@ export class WalletConnectWalletAdapter extends BaseSignerWalletAdapter {
 			const { publicKey } = await this._wallet.connect()
 			this._publicKey = publicKey
 			this.emit('connect', publicKey)
-			this._wallet.client.on('session_delete', this.disconnect)
+			this._wallet.client.on('session_delete', this.disconnect.bind(this))
```

Comment:
https://github.com/WalletConnect/walletconnect-solana-adapter/pull/37/files#diff-ff45047701b988d5794b8985dea7ddfb5f32e51ebef6474fa03fd257867744aaR95

hi ,you need off event after disconnect,like my code

Reply:
@nieheyong would you like to push against this PR?

Reply:
how, can you reopen this pr? branch not delete

Reply:
maybe you can directly update this pr by use code  in file https://github.com/WalletConnect/walletconnect-solana-adapter/pull/37/files#diff-ff45047701b988d5794b8985dea7ddfb5f32e51ebef6474fa03fd257867744aa

Reply:
I'd prefer if we use direct binding instead of creating a new variable to avoid extra code. Will add it here 👍